src/failureAdapters/adapter.js

```diff
@@ -0,0 +1,30 @@
+/**
```

Comment:
I'll review this with time, but by looking at the code, I realized some things. The `FailureAdapter` has to be initialized upfront in DSQ; it would be nice to have some async methods to establish connections and cleanup resources.

Reply:
True. I was thinking of keeping that entirely internal to the failure adapter, which would work in the case of the Kafka failure adapter (provided that KafkaJS exposed events for when the main consumer connects/disconnects), but that won't necessarily work for all types of failure adapters, so probably it should have some initialization and teardown methods.